### PR TITLE
feat: add spine doc

### PR DESCRIPTION
**Pull request by johanzhu**
### Please check if the PR fulfills these requirements

- [ ] The commit message follows our [guidelines](https://github.com/galacean/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## Summary by CodeRabbit

- **New Features**
	- Enhanced documentation for integrating Spine animations, including detailed asset management and usage instructions.
	- Introduction of new examples showcasing Spine animations, including character customization and interactive features.
	- New classes and components added, enabling more complex animation handling and user interactions.

- **Bug Fixes**
	- Improved resource loading efficiency by consolidating multiple URLs into single resource files.

- **Documentation**
	- Comprehensive updates to clarify and enhance the usability of Spine animation integration within the Galacean editor, including new guides and example files.

- **Refactor**
	- Updated rendering methods for Spine animations, improving performance and simplifying entity management.

- **Style**
	- Adjusted camera settings and animation scaling for better visual representation.

### CODE REVIEW

File: docs/zh/graphics/2D/spine/editor.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+order: 1
+title: 在编辑器中使用
+type: 图形
+group: Spine
+label: Graphics/2D/Spine/editor
+---
+
+本章节为大家介绍如何在 Galacean 编辑器中使用 Spine 动画。
+
+Galacean 编辑器内置了对 Spine 动画的支持，无需额外下载或配置，开箱即用，大大简化了开发流程。
```
Comment (cptbtptpbcptdtptp) [inline docs/zh/graphics/2D/spine/editor.md:11]: 顺序换下：
Galacean 编辑器内置了对 Spine 动画的支持，无需额外下载或配置，开箱即用，大大简化了开发流程。本章节为大家介绍如何在 Galacean 编辑器中使用 Spine 动画。

File: docs/zh/graphics/2D/spine/editor.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+## 1. 从 Spine 编辑器导出资产
+第一步，需要从 Spine 编辑器导出你的 Spine 动画素材，你可以在《Spine用户指南》中找到完整的步骤, 说明如何：
```
Comment (cptbtptpbcptdtptp) [inline docs/zh/graphics/2D/spine/editor.md:14]: 《Spine用户指南》给个跳转

Comment (cptbtptpbcptdtptp) [reply docs/zh/graphics/2D/spine/editor.md:14]: 对 Spine 编辑器的版本没有要求吗。

File: docs/zh/graphics/2D/spine/runtime.md [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+// 设置动画资源
+spine.resource = spineResource;
+// 添加至场景
+root.addChild(spineEntity);
+```
+
+
+## 使用运行时 API 
+
+在[前一个章节]()中，为大家介绍了编辑器中 Spine 组件的配置项。通过这些配置项目，能够对 Spine 动画进行一些基础的配置。
```
[the remark this answers is not in the document]
Comment (cptbtptpbcptdtptp) [reply docs/zh/graphics/2D/spine/runtime.md:149]: 同意兔子的说法

File: docs/zh/graphics/2D/spine/runtime.md [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+
+若要对 Spine 动画进行更复杂的操作，还需要在[脚本](https://antg.antgroup.com/engine/docs/latest/cn/script)中使用运行时 API。
+
+Spine 组件暴露了两个重要的对象 (AnimationState 和 Skeleton) 来进行动画控制与骨架操作，以实现更加复杂的效果。下面是详细的使用方式：
+
+#### 动画控制 AnimationState
```
[the remark this answers is not in the document]
Comment (cptbtptpbcptdtptp) [reply docs/zh/graphics/2D/spine/runtime.md:155]: 同意兔子的说法

File: docs/zh/graphics/2D/spine/runtime.md [hunk 5]
```diff
@@ -0,0 +1,382 @@
...
+class YourAmazingScript {
+
+  onStart() {
+    const spine = this.entity.getComponent(SpineAnimationRenderer);
+    const { state } = spine; // AnimationState 对象
+  }
+  
+}
+```
+##### **播放动画**
```
Comment (cptbtptpbcptdtptp) [inline docs/zh/graphics/2D/spine/runtime.md:167]: 同上

File: docs/zh/graphics/2D/spine/runtime.md [hunk 6]
```diff
@@ -0,0 +1,382 @@
...
+type: 图形
+group: Spine
+label: Graphics/2D/Spine/runtime
+---
+
+本章节为大家介绍如何在代码中使用 Galacean Spine 运行时。
+
+无论是通过编辑器导出的项目，或者 procode 项目，都需要通过安装 @galacean/engine-spine (即Galacean Spine 运行时) 来实现 Spine 动画的加载与渲染。
+
+## 安装
```
Comment (cptbtptpbcptdtptp) [inline docs/zh/graphics/2D/spine/runtime.md:13]: 「无论是通过编辑器导出的项目，或者 procode 项目，都需要通过安装 @galacean/engine-spine (即Galacean Spine 运行时) 来实现 Spine 动画的加载与渲染。」
放到这

File: docs/zh/graphics/2D/spine/editor.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+1. [导出 skeleton 和 animation 数据](https://zh.esotericsoftware.com/spine-export)
+2. [导出包含 skeleton 图像的 texture atlases](https://zh.esotericsoftware.com/spine-texture-packer)
```
[the remark this answers is not in the document]
Comment (johanzhu) [reply docs/zh/graphics/2D/spine/editor.md:18]: done

File: docs/zh/graphics/2D/spine/editor.md [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+6. 将会得到三个如下文件：
+
+<img src="https://intranetproxy.alipay.com/skylark/lark/0/2024/png/76063/1721206737659-cd768958-f4e4-4edd-a6ff-0bd16bd82d22.png#clientId=uf07edf19-371c-4&from=paste&height=152&id=uf58744da&originHeight=246&originWidth=656&originalType=binary&ratio=2&rotation=0&showTitle=false&size=70043&status=done&style=shadow&taskId=u31c50a47-bd13-4c0a-80a4-76db96d8118&title=&width=406" width="406">
+
+spineboy.skel 包含了 skeleton  animation 数据，spineboy.atlas 包含了 texture atlas 信息，导出的图片可能有多张，每张图片都代表了 texture altas 中的一页
+
+## 2. 在 Galacean 编辑器中导入资产
+从 Spine 编辑器导出资产后，第二步就要将资产导入至 Galacean 编辑器了。打开编辑器后，将导出的文件直接拖入到[资产面板](https://antg.antgroup.com/engine/docs/latest/cn/assets-interface)中，即可完成上传
+
+<img src="https://intranetproxy.alipay.com/skylark/lark/0/2024/gif/76063/1721208067896-206d850e-20fc-44a6-9987-6a8556b10e3b.gif#clientId=uf07edf19-371c-4&from=paste&height=540&id=u4bb88103&originHeight=1080&originWidth=1984&originalType=binary&ratio=2&rotation=0&showTitle=false&size=8620071&status=done&style=shadow&taskId=ua9290849-8cb2-46ac-98b6-e2b22ab0d83&title=&width=992" width="992">
```
[the remark this answers is not in the document]
Comment (johanzhu) [reply docs/zh/graphics/2D/spine/editor.md]: done

File: docs/zh/graphics/2D/spine/editor.md [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+
+## 3. 添加 Spine 动画
+
+完成资产上传后，第三步，需要将 Spine 动画添加至场景中。一共有三种方式：
+
+1. 拖入添加
+
+拖入添加是最快捷的一种方式。点击 SpineSkeletonData 资产，按住后拖动到视图区，就能快速创建一个添加了 Spine 组件的实体，并指定资产为刚刚选中的 SpineSkeletonData 资产。
+
+<img src="https://intranetproxy.alipay.com/skylark/lark/0/2024/gif/76063/1721210070086-6db51cd8-4493-4b4d-93ee-16c8187fe843.gif#clientId=uf07edf19-371c-4&from=paste&height=540&id=ucc4cc750&originHeight=1080&originWidth=1984&originalType=binary&ratio=2&rotation=0&showTitle=false&size=7335004&status=done&style=shadow&taskId=u97b775ac-f0f4-4f0f-80f6-077cacc66d5&title=&width=992" width="992">
```
[the remark this answers is not in the document]
Comment (johanzhu) [reply docs/zh/graphics/2D/spine/editor.md]: done

File: docs/zh/graphics/2D/spine/editor.md [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+<img src="https://intranetproxy.alipay.com/skylark/lark/0/2024/png/76063/1721269154975-11ceb229-2142-4c6c-9552-0afba11ae924.png#clientId=uf07edf19-371c-4&from=paste&height=476&id=u450ac092&originHeight=1460&originWidth=1048&originalType=binary&ratio=2&rotation=0&showTitle=false&size=276525&status=done&style=shadow&taskId=u1924fec4-712d-4f69-b343-2ab8021cc8e&title=&width=342" width="342">
+
+添加了 Spine 组件后，同样需要指定组件的 Resource，也就是 Spine 组件要渲染的 SpineSkeletonData 资产。
+
+### Spine 组件配置
+以上三种添加 Spine 动画的方法实际上本质其实是相同的，都是通过给实体 `添加 Spine 组件` ，来让 Spine 动画添加至场景中的。
+
+Spine 组件的配置如下：
+
+<img src="https://intranetproxy.alipay.com/skylark/lark/0/2024/png/76063/1721269455151-ceaf0b7e-a00b-4ba4-a109-0cdb08c135f2.png#clientId=uf07edf19-371c-4&from=paste&height=199&id=ucac2f7de&originHeight=398&originWidth=1060&originalType=binary&ratio=2&rotation=0&showTitle=false&size=82934&status=done&style=shadow&taskId=ub50cded7-5559-41de-b104-9db3c4bb4f5&title=&width=530" width="503">
```
[the remark this answers is not in the document]
Comment (johanzhu) [reply docs/zh/graphics/2D/spine/editor.md]: done

File: docs/zh/graphics/2D/spine/runtime.md [hunk 11]
```diff
@@ -0,0 +1,382 @@
...
+await engine.resourceManager.load({
+  url: projectInfo.url,
+  type: AssetType.Project,
+})
+```
+
+<b>若未添加至场景中，则需要在代码中手动加载</b>，步骤如下：
+1. 首先，需要找到 Spine 动画的资产链接，点击 Galacean 编辑器的下载按钮，选择 project URL，拷贝 project.json 后打开，找到上传的 spine 动画文件（skel / json）：
+
+<img src="https://intranetproxy.alipay.com/skylark/lark/0/2024/png/76063/1721297144951-6dafb4e3-ccfa-495e-a540-8b3918b66400.png#clientId=u4b0ad8a6-bdc5-4&from=paste&height=480&id=u8ed78f33&originHeight=533&originWidth=359&originalType=binary&ratio=1&rotation=0&showTitle=false&size=40311&status=done&style=none&taskId=udf9587bf-1818-470f-a108-8ca0d7fe9d0&title=&width=323" width="330">
```
[the remark this answers is not in the document]
Comment (johanzhu) [reply docs/zh/graphics/2D/spine/runtime.md]: done

File: docs/zh/graphics/2D/spine/runtime.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (johanzhu) [reply docs/zh/graphics/2D/spine/runtime.md]: done

File: docs/zh/graphics/2D/spine/runtime.md [hunk 12]
```diff
@@ -0,0 +1,382 @@
...
+    mixAndMatchSkin.addSkin(skeletonData.findSkin("clothes/hoodie-orange"));
+    mixAndMatchSkin.addSkin(skeletonData.findSkin("legs/pants-jeans"));
+    mixAndMatchSkin.addSkin(skeletonData.findSkin("accessories/bag"));
+    mixAndMatchSkin.addSkin(skeletonData.findSkin("accessories/hat-red-yellow"));
+    this.skeleton.setSkin(mixAndMatchSkin);
+  }
+
+}
+```
+代码中皮肤的名称来自于 mix-and-match 示例。
```
[the remark this answers is not in the document]
Comment (johanzhu) [reply docs/zh/graphics/2D/spine/runtime.md]: done

File: docs/zh/graphics/2D/spine/runtime.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (johanzhu) [reply docs/zh/graphics/2D/spine/runtime.md]: done

File: docs/zh/graphics/2D/spine/editor.md (hunk 1 above)
Comment (johanzhu) [reply docs/zh/graphics/2D/spine/editor.md]: done

File: docs/zh/graphics/2D/spine/editor.md (hunk 2 above)
Comment (johanzhu) [reply docs/zh/graphics/2D/spine/editor.md]: done

File: docs/zh/graphics/2D/spine/runtime.md (hunk 5 above)
Comment (johanzhu) [reply docs/zh/graphics/2D/spine/runtime.md]: done

File: docs/zh/graphics/2D/spine/runtime.md (hunk 6 above)
Comment (johanzhu) [reply docs/zh/graphics/2D/spine/runtime.md:12]: done

**Review by cptbtptpbcptdtptp — APPROVED** [review]
+1

**Pull request merged**